sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -206,6 +206,22 @@ object DateTimeUtils {
     (day.toInt, micros * 1000L)
   }
 
+  /*
+   * Converts the timestamp to milliseconds since epoc. In spark timestamp values have microseconds
+   * precision, so this conversion is lossy.
+   */
+  def toMillis(us: SQLTimestamp): Long = {
+    val julian_us = us + JULIAN_DAY_OF_EPOCH * MICROS_PER_DAY
```

Comment:
@davies Thank you for your  comments Davies. Per Lian's comment in the jira,
https://issues.apache.org/jira/browse/SPARK-8824
parquet 1.8 which is what we are using currently does not have support for TIMESTAMP_MICROS yet. He suggested we implement TIMESTAMP_MILLIS for now.